### PR TITLE
Unsignedness Checker Code Review

### PR DESCRIPTION
**Pull request by cmackie**

### CODE REVIEW

File: framework/tests/all-systems/GetClassTest.java [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
     // See AnntoatedTypeFactory.adaptGetClassReturnTypeToReceiver
 
+    //@SuppressWarnings("unsignedness")
```
Comment (mernst) [inline framework/tests/all-systems/GetClassTest.java]: If this is commented out, please remove it to avoid irrelevant changes in the diff.

**Comment by mernst** [comment]
The Travis tests are passing under JDK 8 but failing under JDK 7, because some code uses Java-8-specific interfaces.

We have a policy of not doing code reviews unless the tests are passing.

Can you get the tests passing or ask for advice about how to do that?

**Comment by mernst** [comment]
Can you please pull master into your fork and resolve the conflicts?

File: checker/src/org/checkerframework/checker/unsignedness/UnsignednessAnnotatedTypeFactory.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+ * @checker_framework.manual #unsignedness-checker Unsignedness Checker
+ */
+public class UnsignednessAnnotatedTypeFactory extends BaseAnnotatedTypeFactory{
+
+    private final AnnotationMirror UNSIGNED;
+    private final AnnotationMirror SIGNED;
+    private final AnnotationMirror UNKNOWN_SIGNEDNESS;
+
+    // These are commented out until issues with making boxed implicitely signed
+    // are worked out.
```
Comment (smillst) [inline checker/src/org/checkerframework/checker/unsignedness/UnsignednessAnnotatedTypeFactory.java]: Before this is merged, an issue needs to be opened explaining what the problems are with boxed primitives, and the issue should be referenced here.

**Comment by smillst** [comment]
I have reviewed the code and made a few changes (https://github.com/cmackie/checker-framework/pull/7).

File: checker/manual/unsignedness-checker.tex [hunk 3]
```diff
@@ -0,0 +1,144 @@
+\htmlhr
+\chapter{Unsignedness Checker\label{unsignedness-checker}}
```
Comment (smillst) [inline checker/manual/unsignedness-checker.tex]: The manual should mention UnsignednessUtil.

**Comment by smillst** [comment]
This checker needs to be added to the Eclipse plugin.  We don't have documentation on who to do this, so I'll add this checker then write up the documentation.

File: checker/manual/unsignedness-checker.tex (hunk 3 above)
Comment (mernst) [inline checker/manual/unsignedness-checker.tex]: Done.

**Comment by smillst** [comment]
merged via ef4f225

**Pull request closed without merging**